metricbeat/module/elasticsearch/pending_tasks/data.go

```diff
@@ -0,0 +1,44 @@
+package pending_tasks
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+)
+
+var (
+	schema = s.Schema{
+		"insert_order":         c.Int("insert_order"),
+		"priority":             c.Str("priority"),
+		"source":               c.Str("source"),
+		"time_in_queue_millis": c.Int("time_in_queue_millis"),
+		"time_in_queue":        c.Str("time_in_queue"),
+	}
+)
+
+func eventsMapping(content []byte) ([]common.MapStr, error) {
+	tasksStruct := struct {
+		Tasks []map[string]interface{} `json:"tasks"`
+	}{}
+
+	if err := json.Unmarshal(content, &tasksStruct); err != nil {
+		return nil, err
+	}
+
+	var events []common.MapStr
+	errors := s.NewErrors()
+
+	for _, task := range tasksStruct.Tasks {
+		event, errs := eventMapping(task)
+		errors.AddErrors(errs)
+		events = append(events, event)
+	}
+
+	return events, errors
```

Comment:
Maybe sending all pending tasks is too much for heavy loaded clusters, it could be useful also to a summarized event with information like number of events or max time in queue, wdyt?

Reply:
I agree with you.

We could just send an event with following metrics : 
- total pending tasks
- total pending tasks with priority urgent
- total pending tasks with priority high
- max time in queue

It's a good indicator.

Nonetheless we lose the source field. I'll have to request the cluster what's happening.

Reply:
See comment above, i think we should have two different metricsets if that works.